src/Raven.Server/Documents/Subscriptions/Sharding/ShardSubscriptionStorage.cs

```diff
@@ -0,0 +1,56 @@
+﻿using Raven.Client.Documents.Subscriptions;
+using Raven.Client.Exceptions.Documents.Subscriptions;
+using Raven.Client.Json.Serialization;
+using Raven.Client.ServerWide;
+using Raven.Server.Documents.Sharding;
+using Raven.Server.ServerWide;
+using Raven.Server.ServerWide.Context;
+
+namespace Raven.Server.Documents.Subscriptions.Sharding;
+
+public class ShardSubscriptionStorage : SubscriptionStorage
+{
+    public ShardSubscriptionStorage(ShardedDocumentDatabase db, ServerStore serverStore) : base(db, serverStore)
+    {
+    }
+
+    public override void HandleDatabaseRecordChange(DatabaseRecord databaseRecord)
```

Comment:
what is different here from the base class?

Reply:
you only care about the responsible node for the subscription here

Reply:
okay, so if I disable or change the subscription's query, it will be handled by the orchestrator to drop this one?